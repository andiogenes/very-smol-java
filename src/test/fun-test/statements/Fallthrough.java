class Main {
    int x = 0;

    void fallthrough() {
        switch (x) {
            case 0:
                println(10);
            case 1:
                println(20);
            case 2:
                println(30);
            default:
                println(40);
        }
        println();
    }

    void main() {
        x = 0;
        fallthrough();

        x = 1;
        fallthrough();

        x = 2;
        fallthrough();

        x = 3;
        fallthrough();
    }
}