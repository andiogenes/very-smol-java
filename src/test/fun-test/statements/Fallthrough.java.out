x : INT = 0 : INT
x : INT = 0 : INT
10
20
30
40

x : INT = 1 : INT
20
30
40

x : INT = 2 : INT
30
40

x : INT = 3 : INT
40

